### PR TITLE
Run `tsc` as part of the lint script and upgrade dependencies

### PR DESCRIPTION
**Pull request by sandhose**

### CODE REVIEW

File: src/global.d.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2023 New Vector Ltd.
+Copyright 2023  New Vector Ltd
```
Comment (t3chguy) [inline src/global.d.ts:2]: This looks wrong?

File: tsconfig.json [hunk 2]
```diff
@@ -1,33 +1,11 @@
...
-    "typeRoots": ["./node_modules/@types", "./src/@types"],
-    "types": ["node"],
-
-    /* Completeness */
-    "skipLibCheck": true /* Skip type checking all .d.ts files. */
-  }
+    "types": ["vite/client"]
+  },
+  "include": ["src"],
+  "exclude": ["**/*.stories.tsx"],
```
Comment (t3chguy) [inline tsconfig.json:9]: Does this not mean errors like the one in your recent PR will continue to be permissable?

File: src/global.d.ts (hunk 1 above)
Comment (sandhose) [reply src/global.d.ts]: Woops, I actually re-created that file and copy-pasted the copyright from the wrong place. Fixed

File: tsconfig.json (hunk 2 above)
Comment (sandhose) [reply tsconfig.json]: Hah, this was me trying to figure out why the `vite-plugin-dts` was not working as expected and I forgot to remove it. Fixed

File: .storybook/main.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ const config: StorybookConfig = {
     "@storybook/addon-essentials",
     "@storybook/addon-interactions",
     "@storybook/addon-designs",
-    "@storybook/addon-styling",
```
Comment (sandhose) [inline .storybook/main.ts:10]: `addon-styling` is deprecated

File: .storybook/main.ts [hunk 4]
```diff
@@ -22,11 +22,6 @@ const config: StorybookConfig = {
   viteFinal: (config) => {
     return {
       ...config,
-
-      // In the `vite.config.ts` file, we set the `experimental.renderBuiltUrl`
-      // option, which breaks the storybook build. This clears that option.
-      experimental: undefined,
```
Comment (sandhose) [inline .storybook/main.ts:28]: That was an unnecessary leftover

File: package.json [hunk 5]
```diff
@@ -50,58 +51,58 @@
...
     "vite-plugin-svgr": "^3.2.0",
-    "vitest": "^0.33.0"
+    "vitest": "^0.34.4"
   },
   "dependencies": {
     "@radix-ui/react-form": "^0.0.3",
     "@radix-ui/react-tooltip": "^1.0.6",
     "classnames": "^2.3.2",
-    "graphemer": "^1.4.0",
-    "rimraf": "^3.0.1"
```
Comment (sandhose) [inline package.json:104]: No idea why it was in the package dependencies and not devDependencies

File: package.json [hunk 6]
```diff
@@ -118,5 +119,8 @@
     "@types/react-dom": {
       "optional": true
     }
+  },
+  "resolutions": {
+    "jackspeak": "2.1.1"
```
Comment (sandhose) [inline package.json:124]: Workaround for https://github.com/storybookjs/storybook/issues/22431#issuecomment-1630086092

**Pull request merged**